File: application/src/main/java/whenyourcar/application/converter/user/UserLikeConverter.java
package whenyourcar.application.converter.user;

import whenyourcar.domain.entity.Car;
import whenyourcar.domain.entity.User;
import whenyourcar.domain.entity.UserLike;
import org.springframework.data.domain.Page;
import org.springframework.stereotype.Component;
import whenyourcar.domain.query.car.SearchCarsQuery;
import whenyourcar.application.dto.user.like.UserLikeResponse;

@Component
public class UserLikeConverter {
    public UserLike toUserLike(Car car, User user){
        return UserLike.builder()
                .car(car)
                .user(user)
                .build();
    }

    public UserLikeResponse.UserLikesResponseDto toUserLikes(Page<SearchCarsQuery> searchCarsQueries, Long userId) {
        return UserLikeResponse.UserLikesResponseDto.builder()
                .userId(userId)
                .searchCarsQueries(searchCarsQueries)
                .build();
    }
}

File: application/src/main/kotlin/howmuchcar/application/converter/user/UserLikeConverter.kt
package howmuchcar.application.converter.user

import howmuchcar.application.dto.user.UserLikeResponse
import howmuchcar.domain.entity.Car
import howmuchcar.domain.entity.User
import howmuchcar.domain.entity.UserLike
import howmuchcar.domain.query.car.SearchCarsQuery
import howmuchcar.domain.query.user.SearchCarLikeQuery
import org.springframework.data.domain.Page
import org.springframework.stereotype.Component

@Component
class UserLikeConverter {
    fun toUserLike(car: Car, user: User) : UserLike {
        return UserLike.builder()
            .car(car)
            .user(user)
            .build()
    }

    fun toUserLikes(searchCarsQueries: Page<SearchCarLikeQuery>, userId: Long): UserLikeResponse {
        return UserLikeResponse(
            userId = userId,
            userLikeCars = searchCarsQueries
        )
    }
}